pkg/optjson/optjson_test.go

```diff
@@ -88,3 +88,175 @@ func TestString(t *testing.T) {
 		}
 	})
 }
+
+func TestSlice(t *testing.T) {
+	t.Run("slice of ints", func(t *testing.T) {
+		cases := []struct {
+			data      string
+			wantErr   string
+			wantRes   Slice[int]
+			marshalAs string
+		}{
+			{`[1,2,3]`, "", SetSlice([]int{1, 2, 3}), `[1,2,3]`},
+			{`[]`, "", SetSlice([]int{}), `[]`},
+			{`null`, "", Slice[int]{Set: true, Valid: false, Value: []int{}}, `null`},
+			{`[1,"2",3]`, "cannot unmarshal string", Slice[int]{Set: true, Valid: false, Value: nil}, `null`},
+			{`123`, "cannot unmarshal number", Slice[int]{Set: true, Valid: false, Value: []int(nil)}, `null`},
+		}
+
+		for _, c := range cases {
+			t.Run(c.data, func(t *testing.T) {
+				var s Slice[int]
+				err := json.Unmarshal([]byte(c.data), &s)
+
+				if c.wantErr != "" {
+					require.Error(t, err)
+					require.ErrorContains(t, err, c.wantErr)
+				} else {
+					require.NoError(t, err)
+				}
+				require.Equal(t, c.wantRes, s)
+
+				b, err := json.Marshal(s)
+				require.NoError(t, err)
+				require.Equal(t, c.marshalAs, string(b))
+			})
+		}
+	})
+
+	t.Run("slice of strings", func(t *testing.T) {
+		cases := []struct {
+			data      string
+			wantErr   string
+			wantRes   Slice[string]
+			marshalAs string
+		}{
+			{`["foo", "bar"]`, "", SetSlice([]string{"foo", "bar"}), `["foo","bar"]`},
+			{`[""]`, "", SetSlice([]string{""}), `[""]`},
+			{`null`, "", Slice[string]{Set: true, Valid: false, Value: []string{}}, `null`},
+			{`["foo", 123]`, "cannot unmarshal number", Slice[string]{Set: true, Valid: false, Value: []string(nil)}, `null`},
+		}
+
+		for _, c := range cases {
+			t.Run(c.data, func(t *testing.T) {
+				var s Slice[string]
+				err := json.Unmarshal([]byte(c.data), &s)
+
+				if c.wantErr != "" {
+					require.Error(t, err)
+					require.ErrorContains(t, err, c.wantErr)
+				} else {
+					require.NoError(t, err)
+				}
+				require.Equal(t, c.wantRes, s)
+
+				b, err := json.Marshal(s)
+				require.NoError(t, err)
+				require.Equal(t, c.marshalAs, string(b))
+			})
+		}
+	})
+
+	t.Run("slice of bools", func(t *testing.T) {
+		cases := []struct {
+			data      string
+			wantErr   string
+			wantRes   Slice[bool]
+			marshalAs string
+		}{
+			{`[true, false]`, "", SetSlice([]bool{true, false}), `[true,false]`},
+			{`[true, "false"]`, "cannot unmarshal string", Slice[bool]{Set: true, Valid: false, Value: []bool(nil)}, `null`},
+			{`null`, "", Slice[bool]{Set: true, Valid: false, Value: []bool{}}, `null`},
+		}
+
+		for _, c := range cases {
+			t.Run(c.data, func(t *testing.T) {
+				var s Slice[bool]
+				err := json.Unmarshal([]byte(c.data), &s)
+
+				if c.wantErr != "" {
+					require.Error(t, err)
+					require.ErrorContains(t, err, c.wantErr)
+				} else {
+					require.NoError(t, err)
+				}
+				require.Equal(t, c.wantRes, s)
+
+				b, err := json.Marshal(s)
+				require.NoError(t, err)
+				require.Equal(t, c.marshalAs, string(b))
+			})
+		}
+	})
+}
+
+func TestSliceWithinStruct(t *testing.T) {
```

Comment:
For the cases in this test, I think it would be helpful to have the struct field names in the literals to improve readability.

Reply:
While I don't agree, I added them https://github.com/fleetdm/fleet/pull/14190/commits/648dc853a4f2cd36f8a42ad8250374b8aa060f42 in as I can see how it can help other people